### DISCUSSION TITLE
Moving tests on Loaclstack

### DISCUSSION THREAD

**Discussion by semenodm** [laserdisc-io/fs2-aws#592] [Ideas]
I found it would be useful to run all tests agains Loaclstack instead of mocking AWS SDK, KPL, KCL, especially KCL.
However i tried to reproduce stream resharding and found that Localstack doesn't support UpdateShardCount operation, it is not failing, but it does nothing. 
I did some search and found following discussions
1. https://github.com/localstack/localstack/issues/654
2. Looks like this is the stub to support API call but it actually does nothing 
      https://github.com/localstack/localstack/pull/1229
3. https://github.com/mhart/kinesalite/issues/61
4. kinesalite is the project used to support Kinesis integration in Localstack, this PR is still open, but looks like feature is implemented
     https://github.com/mhart/kinesalite/pull/48

@JohnG84 @barryoneill

**Comment by JohnG84** [comment]
Sounds like a good idea to me. So https://github.com/mhart/kinesalite/pull/103 (which appears to have been open for over 5 months) needs to be merged, then localstack needs to upgrade to the latest version of kinesalite (and some minor code changes are required)?

**Reply by semenodm** [reply]
yeah, i dropped some notes there, to demonstrate what is the problem, hopefully someone merge this work and then we need to remove that stub in Localstack :)

**Comment by etspaceman** [comment]
Just following up here. Localstack now uses [kinesis-mock](https://github.com/etspaceman/kinesis-mock), which has this support.
